java/vector/src/main/java/org/apache/arrow/vector/table/BaseRow.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.table;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Provides row based access to the data held by a {@link Table}.
+ *
+ * <p>
+ * This API is EXPERIMENTAL.
+ */
+public abstract class BaseRow {
```

Comment:
For the class hierarchy: is a hypothetical MutableRow expected to inherit from BaseRow (in which case there's not a great way to abstract over immutable/mutable rows for read-only access) or from Row (in which case BaseRow seems a little redundant)?

Reply:
Since things are experimental, I think we can keep it for now and see how the mutable implementation actually looks